data/pathfinder/paizo/roleplaying_game/core_rulebook/cr_abilities_class.lst

```diff
@@ -83,7 +83,7 @@ Fighter			CATEGORY:Class		DEFINE:Fighter_CFP_Level|0														ABILITY:Intern
 Monk				CATEGORY:Class		DEFINE:Monk_CFP_Level|0	DEFINE:Pool_Monk_Class_Selection|0									ABILITY:Class|AUTOMATIC|Monk ~ Standard Class|!PREABILITY:1,CATEGORY=CLASS,TYPE.AltMonkChoice															BONUS:VAR|Monk_CFP_Level|MonkLVL|TYPE=Base
 Paladin			CATEGORY:Class		DEFINE:Paladin_CFP_Level|0														ABILITY:Internal|AUTOMATIC|Class Skills ~ Paladin|PREVAREQ:Paladin_CF_ClassSkills,0																	BONUS:VAR|Paladin_CFP_Level|PaladinLVL|TYPE=Base
 Ranger			CATEGORY:Class		DEFINE:Ranger_CFP_Level|0														ABILITY:Internal|AUTOMATIC|Class Skills ~ Ranger|PREVAREQ:Ranger_CF_ClassSkills,0																	BONUS:VAR|Ranger_CFP_Level|RangerLVL|TYPE=Base
-Rogue				CATEGORY:Class		DEFINE:Rogue_CFP_Level|0	DEFINE:Pool_Rogue_Class_Selection|0								ABILITY:Internal|AUTOMATIC|Class Skills ~ Rogue|PREVAREQ:Rogue_CF_ClassSkills,0	ABILITY:Class|AUTOMATIC|Rogue ~ Standard Class|!PREABILITY:1,CATEGORY=CLASS,TYPE.AltRogueChoice	BONUS:VAR|Rogue_CFP_Level|RogueLVL|TYPE=Base	BONUS:VAR|RogueSneakAttackLVL|RogueLVL	BONUS:VAR|TrapfindingLVL|RogueLVL	BONUS:VAR|RogueTalentLVL|RogueLVL|PREVARGTEQ:RogueLVL,2	BONUS:VAR|RogueTrapSenseLVL|RogueLVL|PREVARGTEQ:RogueLVL,3	BONUS:VAR|UncannyDodgeFlankingLevel|RogueLVL|TYPE=EachClass.REPLACE|PREVARGTEQ:RogueLVL,4	BONUS:VAR|UncannyDodgeLVL|1|PREVARGTEQ:RogueLVL,4|PREVAREQ:RogueUncannyDodge,0	BONUS:VAR|UncannyDodgeLVL|1|PREVARGTEQ:RogueLVL,8|PREVAREQ:RogueImprovedUncannyDodge,0	BONUS:VAR|MasterStrikeLVL|RogueLVL|PREVARGTEQ:RogueLVL,20
+Rogue				CATEGORY:Class		DEFINE:Rogue_CFP_Level|0	DEFINE:Pool_Rogue_Class_Selection|0								ABILITY:Internal|AUTOMATIC|Class Skills ~ Rogue|PREVAREQ:Rogue_CF_ClassSkills,0	ABILITY:Class|AUTOMATIC|Rogue ~ Standard Class|!PREABILITY:1,CATEGORY=CLASS,TYPE.AltRogueChoice	BONUS:VAR|Rogue_CFP_Level|RogueLVL|TYPE=Base	BONUS:VAR|RogueSneakAttackLVL|RogueLVL	BONUS:VAR|TrapfindingLVL|RogueLVL	BONUS:VAR|RogueTalentLVL|RogueLVL|PREVARGTEQ:RogueLVL,2	BONUS:VAR|RogueTrapSenseLVL|RogueLVL|PREVARGTEQ:RogueLVL,3	BONUS:VAR|MasterStrikeLVL|RogueLVL|PREVARGTEQ:RogueLVL,20
 Sorcerer			CATEGORY:Class		DEFINE:Sorcerer_CFP_Level|0														ABILITY:Internal|AUTOMATIC|Class Skills ~ Sorcerer|PREVAREQ:Sorcerer_CF_ClassSkills,0																BONUS:VAR|Sorcerer_CFP_Level|SorcererLVL|TYPE=Base
```

Comment:
The "Uncanny Dodge Tracker" approach is currently being used with the Barbarian, I more or less just copied that. There are class abilities that are named "Barbarian ~ Uncanny Dodge" and "Barbarian ~ Improved Uncanny Dodge" (same for rogue), but they are never granted to the classes. The two CRB prestige classes that grant uncanny dodge do the same - they just increase the UncannyDodgeLevel variable by 1, which grants the generic Uncanny Dodge. Increasing it by 1 again grants Improved Uncanny Dodge. 
The rogue had the problem that one line just granted uncanny dodge, no check for archetypes whatsoever, so it could never be replaced again.
The not stacking bug came from the fact that the stacking counter was put on the unreferenced class features and never got used. 
The Uncanny Dodges are generally kind of messy, I could certainly do a complete cleanup with properly named class abilities (this would include changing the currently existing and working code for the Barbarian (plus unchained), the Assassin, the Rogue (plus unchained) and another few classes that I don't recall right now.